src/frontend/packages/archipelago-layout/src/layout/DefaultLayout/Layout.js

```diff
@@ -4,31 +4,35 @@ import { makeStyles, Typography } from '@material-ui/core';
 import AppBar from './AppBar';
 import TreeMenu from './TreeMenu/TreeMenu';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles(theme => ({
   appFrame: {
-    marginTop: 56
-  },
-  title: {
-    position: 'absolute',
-    top: 80,
-    textOverflow: 'ellipsis',
-    whiteSpace: 'nowrap',
-    overflow: 'hidden'
+    marginTop: 56,
+    [theme.breakpoints.up('sm')]: {
+      '& #main-content': {
+        paddingTop: 8,
+        paddingLeft: 5
+      }
+    }
   }
-});
+}));
 
-const Layout = ({ appBar, menu, userMenu, children, ...otherProps }) => {
+const Layout = ({ appBar, menu, userMenu, children, labelNbRows, ...otherProps }) => {
   const classes = useStyles();
+  const LayoutTreeMenu = props => <TreeMenu {...props} labelNbRows={labelNbRows} />;
   return (
-    <RaLayout {...otherProps} classes={{ appFrame: classes.appFrame }} appBar={appBar} menu={menu}>
+    <RaLayout
+      {...otherProps}
+      classes={{ appFrame: classes.appFrame }}
+      appBar={appBar}
+      menu={menu ? menu : LayoutTreeMenu}
```

Comment:
Je n'arrive pas à utiliser React.cloneElement dans ce cas de figure.

Si je remplace
`menu={menu ? menu : LayoutTreeMenu}`
par 
`menu={React.cloneElement(menu, {labelNbLines:labelNbLines})}`

j'ai une erreur :
```
Warning: Failed prop type: Invalid prop 'menu' supplied to 'LayoutWithoutTheme': the prop is not a valid React component
[...]
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: <Unknown />. Did you accidentally export a JSX literal instead of a component?
[...]
Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
```

console.log de menu :
```
ƒ TreeMenu() {
  return /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1___default.a.createElement("div", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      col…
```
console.log de React.cloneElement(menu, {labelNbLines:labelNbLines}) :
```
{
$$typeof: Symbol(react.element)
key: undefined
props: {labelNbLines: 2}
ref: undefined
type: undefined
_owner: undefined
_store: {validated: false}
_self: undefined
_source: undefined
[[Prototype]]: Object
}
```

Reply:
Ah c'est sans doute qu'il te faut passer un composant React et non sa définition.
C'est à dire p.ex. `<TreeMenu />` et non `TreeMenu`.
React.cloneElement a besoin de composants (sinon on peut utiliser React.createElement, mais c'est moins pratique pour simplement injecter de nouvelles props à un composant existant, ce qui est ce que tu essaies de faire).

Reply:
J'ai fait quelques tests.
Je crois que le c'est le composant Layout de RA qui attend une prop menu sous forme de définition de composant.

Ca passe avec :
- menu={menu}
- menu={TreeMenu}

Ca plante avec :
- menu={<menu/>}
- menu={<TreeMenu/>}

Or React.cloneElement produit un composant et pas une définition de composant.

Reply:
Ah OK, bon alors on laisse tomber, désolé pour ce mauvais conseil !! Ta manière de faire était la bonne.

Reply:
pas de soucis, c'est formateur !